pyvisa_sim/testsuite/test_common.py

```diff
@@ -6,35 +6,72 @@
 
 
 @pytest.mark.parametrize(
-    "data, mask, send_end, want",
+    "bits, want",
     [
-        (b"1234", None, False, b"1234"),
-        (b"41234", 3, False, b"40004"),
-        # Can't figure this one out. Getting ValueError: bytes must in in range(0, 256)
-        # If I knew more about the purpose of `iter_bytes` then maybe I could
-        # reconcile it, but I don't have time to investigate right now.
-        pytest.param(
-            b"1234",
-            3,
-            True,
-            b"1234",
-            marks=pytest.mark.xfail(
-                reason=(
-                    "ValueError bytes must be in range(0, 256). TODO: figure"
-                    " out correct 'want' value."
-                )
-            ),
-        ),
+        (0, 0b0),
+        (1, 0b1),
+        (5, 0b0001_1111),
+        (7, 0b0111_1111),
+        (8, 0b1111_1111),
+        (11, 0b0111_1111_1111),
+    ],
+)
+def test_create_bitmask(bits, want):
+    got = common._create_bitmask(bits)
+    assert got == want
+
+
+@pytest.mark.parametrize(
+    "data, data_bits, send_end, want",
+    [
+        (b"\x01", None, False, b"\x01"),
+        (b"hello world!", None, False, b"hello world!"),
+        # Only apply the mask
+        (b"\x03", 2, None, b"\x03"),  # 0b0000_0011 --> 0b0000_0011
+        (b"\x04", 2, None, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\xff", 5, None, b"\x1f"),  # 0b1111_1111 --> 0b0001_1111
+        (b"\xfe", 7, None, b"\x7e"),  # 0b1111_1110 --> 0b0111_1110
+        (b"\xfe", 8, None, b"\xfe"),  # 0b1111_1110 --> 0b1111_1110
+        (b"\xff", 9, None, b"\xff"),  # 0b1111_1111 --> 0b1111_1111
+        # Always set highest bit *of data_bits* to 0
+        (b"\x04", 2, False, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\x04", 3, False, b"\x00"),  # 0b0000_0100 --> 0b0000_0000
+        (b"\x05", 3, False, b"\x01"),  # 0b0000_0101 --> 0b0000_0001
+        (b"\xff", 7, False, b"\x3f"),  # 0b1111_1111 --> 0b0011_1111
+        (b"\xff", 8, False, b"\x7f"),  # 0b1111_1111 --> 0b0111_1111
+        # Always set highest bit *of data_bits* to 1
+        (b"\x04", 2, True, b"\x02"),  # 0b0000_0100 --> 0b0000_0010
+        (b"\x04", 3, True, b"\x04"),  # 0b0000_0100 --> 0b0000_0100
+        (b"\x01", 3, True, b"\x05"),  # 0b0000_0001 --> 0b0000_0101
+        (b"\x9f", 7, True, b"\x5f"),  # 0b1001_1111 --> 0b0101_1111
+        (b"\x9f", 8, True, b"\x9f"),  # 0b1001_1111 --> 0b1001_1111
+        # data_bits >8 bits act like data_bits=8, as type(data) is "bytes"
+        # which is limited 8 bits per character.
+        (b"\xff", 9, None, b"\xff"),
+        (b"\xff", 9, False, b"\x7f"),
+        (b"\xff", 9, True, b"\xff"),
+        # send_end should only impact the final byte
+        (b"\x6d\x5c\x25\x25", 4, None, b"\r\x0c\x05\x05"),
+        (b"\x6d\x5c\x25\x25", 4, False, b"\r\x0c\x05\x05"),
+        (b"\x6d\x5c\x25\x25", 4, True, b"\r\x0c\x05\x0d"),
```

Comment:
The fact that the case for  `send_end = None` match the other two tells me that the highest bit is not cleared as I would expect for a non-None `send_end` value.